extensions/binding/org.eclipse.smarthome.binding.homematic/src/main/java/org/eclipse/smarthome/binding/homematic/handler/HomematicThingHandler.java

```diff
@@ -508,12 +509,22 @@ public void handleConfigurationUpdate(Map<String, Object> configurationParameter
     @SuppressWarnings("null")
     @Override
     public synchronized void handleRemoval() {
-        final Bridge bridge = getBridge();
-        if (bridge != null && bridge.getHandler() != null
-                && bridge.getConfiguration().as(HomematicConfig.class).isUnpairOnDeletion()) {
+        final Bridge bridge;
+        final ThingHandler handler;
+
+        if ((bridge = getBridge()) == null || (handler = bridge.getHandler()) == null) {
+            super.handleRemoval();
+            return;
+        }
+
+        final HomematicConfig config = bridge.getConfiguration().as(HomematicConfig.class);
+        final boolean unpairOnDeletion = config.isUnpairOnDeletion() || config.isFactoryResetOnDeletion();
+        final boolean factoryResetOnDeletion = config.isFactoryResetOnDeletion();
```

Comment:
You could fetch `config.isFactoryResetOnDeletion()` only once from the `config` and hold it in a variable.

Reply:
I changed the order so that `factoryResetOnDeletion` is reused when assigning to `unpairOnDeletion`.